src/platform_linux/platform.c

```diff
@@ -112,13 +114,16 @@ platform_buffer_getaddr(const buffer_handle *bh)
 platform_status
 platform_buffer_deinit(buffer_handle *bh)
 {
-   int ret;
-   ret = munmap(bh->addr, bh->length);
-   if (ret) {
-      return CONST_STATUS(errno);
-   }
+   if (bh->addr) {
+      int ret;
+      ret = munmap(bh->addr, bh->length);
+      if (ret) {
+         return CONST_STATUS(errno);
+      }
 
-   bh->addr = NULL;
+      bh->addr = NULL;
+   }
+   bh->length = 0;
```

Comment:
This and L124 are added so that we can validate the state of resulting output `bh` in unit-tests.